ui/src/shared/actions/predicates.ts

```diff
@@ -97,6 +110,46 @@ export const resetPredicateState = (): SetPredicateToDefault => ({
   type: 'SET_PREDICATE_DEFAULT',
 })
 
+interface SetKeysByBucket {
+  type: 'SET_KEYS_BY_BUCKET'
+  keys: string[]
+}
+
+const setKeys = (keys: string[]): SetKeysByBucket => ({
+  type: 'SET_KEYS_BY_BUCKET',
+  keys,
+})
+
+export const setBucketAndKeys = (orgID: string, bucketName: string) => async (
+  dispatch: Dispatch<Action>
+) => {
+  const query = `import "influxdata/influxdb/v1"
+  v1.tagKeys(bucket: "${bucketName}")`
+  const keys = await extractBoxedCol(runQuery(orgID, query), '_value').promise
+  dispatch(setBucketName(bucketName))
+  dispatch(setKeys(keys))
+}
+
+interface SetValuesByKey {
```

Comment:
A pattern we have had is that regular actions are at the top and thunks are at the bottom of the document. It would be nice to keep to it maybe, feel not so strongly about this

Reply:
Sounds good. I'll make sure to set the thunks at the bottom and have the regular actions at the top (and alphabetize for sanity)